src/v/cluster/cluster_utils.cc

```diff
@@ -174,13 +174,27 @@ ss::future<> update_broker_client(
 model::broker make_self_broker(const config::node_config& node_cfg) {
     auto kafka_addr = node_cfg.advertised_kafka_api();
     auto rpc_addr = node_cfg.advertised_rpc_api();
+
+    // Calculate memory size
+    const auto shard_mem = ss::memory::stats();
+    uint64_t total_mem = shard_mem.total_memory() * ss::smp::count;
+    uint32_t total_mem_gb = total_mem >> 30;
+
+    // Calculate disk size
+    auto space_info = std::filesystem::space(
+      config::node().data_directory().path);
+    uint32_t disk_gb = space_info.capacity >> 30;
```

Comment:
I'm not clear on why we are trying to cram this into 32 bits, versus storing it elsewhere.

Reply:
It's an existing 32 bit field, which is unfortunately stored in an unversioned serialized structure (broker_properties).  The intent is to pick up things like this in future when we have `serde` versioned encoding everywhere.